Src/Particle/AMReX_Particle.H

```diff
@@ -95,13 +95,47 @@ struct ParticleIDWrapper
     {
         Long r = 0;
 
-        uint64_t sign = m_idata >> 63;  // extract leftmost sign bit
+        uint64_t sign = m_idata >> 63;  // extract uppermost sign bit
         uint64_t val  = ((m_idata >> 24) & 0x7FFFFFFFFF);  // extract next 39 id bits
 
-        Long lval = static_cast<Long>(val);  // bc we take -
+        Long lval = static_cast<Long>(val);  // bc we store the inverse sign bit
         r = (sign) ? lval : -lval;
         return r;
     }
+
+    /** Swap the sign of the id.
+     *
+     * Positive ids become negative and negative ids become positive.
+     */
+    AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+    void negate () noexcept
+    {
+        // The uppermost bit is our id inverse sign.
+        // We will XOR it with 1 to flip it.
+        m_idata ^= static_cast<uint64_t>(1) << 63;
```

Comment:
Need to double check that this is correct:
https://en.wikipedia.org/wiki/Signed_number_representations

Reply:
Unless at the min/max range, we could use something based on https://stackoverflow.com/a/64325162/2719194

but we might need to shift in/out the cpu part again.